From ec9c8a7a2cbb4c1d312a9a897680cbadd922c996 Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Wed, 13 Jun 2018 18:06:40 +0800
Subject: [PATCH 4023/5242] MLK-18592-2 ARM64: dts: imx8qm: update dom node

commit  69360f3ab5b1dbb3b2d601aa507e404a44ca2ba2 from
https://source.codeaurora.org/external/imx/linux-imx.git

Update dom node with name and reg, the two property
will be parsed by xen to create the mapping between
domain name and partition id.

With the mapping, xen could power off the resources
owned by the partition when destroy/reboot/shutdown
the domain.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-mek-dom0.dts     |   17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts
index f089b56..9dec28c 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts
@@ -39,8 +39,23 @@
 		 * SC_R_MU_4A
 		 * The rsrcs and pads will be configured by uboot scu_rm cmd
 		 */
-		dom2: dom@2 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		doma {
 			compatible = "xen,domu";
+			/*
+			 * The name entry in VM configuration file
+			 * needs to be same as here.
+			 */
+			domain_name = "DomU";
+			/*
+			 * The reg property will be updated by U-Boot to
+			 * reflect the partition id.
+			 */
+			reg = <0>;
+			init_on_rsrcs = <
+				SC_R_MU_2A
+			>;
 			rsrcs = <
 				SC_R_GPU_1_PID0
 				SC_R_GPU_1_PID1
-- 
1.7.9.5

